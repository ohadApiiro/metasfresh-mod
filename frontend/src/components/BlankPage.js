import React from 'react';
import PropTypes from 'prop-types';

/**
 * @file Class based component.
 * @module BlankPage
 * @extends Component
 */
const BlankPage = ({ what }) => (
  <div className="blank-page">
    <h1>404</h1>
    <h3>{what ? what + ' not found.' : 'Not found'}</h3>
  </div>
);

/**
 * @typedef {object} Props Component props
 * @prop {string} what
 */
BlankPage.propTypes = {
  what: PropTypes.any,
};

export default BlankPage;
