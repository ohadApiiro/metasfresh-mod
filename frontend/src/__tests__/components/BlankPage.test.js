import BlankPage from '../../components/BlankPage';
import React from 'react';
import renderer from 'react-test-renderer';

test('BlankPage renders correctly', () => {
  const component = renderer.create(<BlankPage />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
