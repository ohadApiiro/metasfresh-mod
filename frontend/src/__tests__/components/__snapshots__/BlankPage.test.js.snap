// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankPage renders correctly 1`] = `
<div
  className="blank-page"
>
  <h1>
    404
  </h1>
  <h3>
    Not found
  </h3>
</div>
`;
